metadata/src/main/java/com/redhat/lightblue/metadata/FieldConstraint.java

```diff
@@ -31,4 +31,10 @@
      * Determines if the constraint is valid for the given field type
      */
     boolean isValidForFieldType(Type fieldType);
+
+
+    /**
+     * Returns the description of the FieldConstraint (for documentation propose)
+     */
+    String getDescription();
```

Comment:
It might need to be specific to each constraint.  Numeric value constraints (minimum, maximum, minLength, etc) will be a number in json (no quotes) vs a pattern which will be a string (with quotes).  I haven't reviewed how this is used yet to see.  Interestingly ArrayElementIdConstraint returns a bunch of text, so I assume given your comment this is broken or it isn't used in a generic way.